### PR TITLE
Switch to updated Spike version

### PR DESCRIPTION
**Pull request by vogelpi**
This commit switches to an updated Spike version with support for ePMP.

~I suspect that this PR won't pass CI because the updated Spike version is not yet available on the tool NAS. I just created a new `ibex-cosim-v0.3` tag for that new version.~ ~Related to that I just realized that I probably made mistake when rebasing as I replaced previous call we had to `get_max_xlen()` with `get_xlen()`. I only discovered now that `get_max_xlen()` has been moved to `isa_parser.h` within Spike. I need fix this.~ This is now done.

This is related to #1579.

### CODE REVIEW

File: dv/cosim/spike_cosim.cc
Comment (vogelpi) [inline dv/cosim/spike_cosim.cc:28]: I am not sure if this is the "correct" way of doing this. I've tried using a `unique_ptr` type for this but if I understand correctly, this doesn't work because we need to pass `isa_parser *` to constructor of `processor` (part of Spike), casting doesn't work and I think it also shouldn't work here. Anyway, some input on this is appreciated.

Comment (rswarbrick) [reply dv/cosim/spike_cosim.cc:28]: There are two clean ways to do this.

First (and most similar to what you've got), turn `isa_parser` into a `unique_ptr<isa_parser_t>` and use `isa_parser.get()` to get the underlying pointer.

Second (and probably less effort), make `isa_parser` an actual `isa_parser_t` as a field and call its constructor in the class initialisation list. Then you can pass it to the `processor_t` constructor with just `&isa_parser`.

Comment (vogelpi) [reply dv/cosim/spike_cosim.cc]: Thanks Rupert, I've now switched to the first option :-)

**Comment by vogelpi** [comment]
Thank you both, I've now fixed the lint error. Let's see if CI passes.

**Comment by vogelpi** [comment]
Cool, CI is passing now. Should we merge this?

**Review by rswarbrick — COMMENTED** [review]
Merging this will cause a "flag day" for anyone who uses Ibex because they'll need to update to version 0.3 to keep working. How difficult would it be to add an ifdef like I did in 96d8aa6c1522eae45b4ec0d03424c9c753544fe7? The idea is that we'd keep both versions working for a week or so to allow things like bisecting across the boundary.

**Comment by vogelpi** [comment]
The argument for enabling bisecting across the version boundary convinced me @rswarbrick . I've now added a similar switch. By default, we continue to use `ibex-cosim-v0.2` for the moment. Feel free to merge this if you're happy with it. I will be OoO for the rest of the week.

**Comment by rswarbrick** [comment]
Thanks for doing this, Pirmin. I'm not completely sure about the `ifdef` though: the `MSTATUSH_GVA` symbol is defined for `ot-v0.2` and not for `ot-v0.1`. To do the same thing, I think you'd need to find a preprocessor symbol that's defined for `ot-v0.3` but not for `ot-v0.2`. If this isn't possible, never mind. But we could always add a version header to our Spike fork for future situations like this.

**Comment by vogelpi** [comment]
Ah, I simply thought `MSTATUS_GVA` was something you added and wasn't aware it's something defined by Spike. I'll need to have a look if there is something similar for `ot-v0.3`. What is definitely working is to add a `-DMSECCFG_GVA` argument for the used simulator in `dv/uvm/core_ibex/yaml/rtl_simulation.yaml`. But it would be much better if this was defined by Spike already.

**Comment by vogelpi** [comment]
I've now identified another define that can be found in v0.3 but not in v0.2 and updated the PR accordingly. Successfully tested in CI (still using v0.2) and locally (using v0.3).

**Review by rswarbrick — APPROVED** [review]
Neat! Let's hope the Spike maintainers keep adding defines :-p

**Pull request merged**
